lib/java/com/google/android/material/textfield/TextInputLayout.java

```diff
@@ -1051,7 +1051,7 @@ private void setEditText(EditText editText) {
 
     final int editTextGravity = this.editText.getGravity();
     collapsingTextHelper.setCollapsedTextGravity(
-        Gravity.TOP | (editTextGravity & ~Gravity.VERTICAL_GRAVITY_MASK));
+        Gravity.TOP | (Gravity.START & ~Gravity.VERTICAL_GRAVITY_MASK));
```

Comment:
We'd have to check with our designers to see what our intended behavior is here. If they want to support other gravities for the EditText, we wouldn't want to hardcode the label's gravity.

I'll ask for design input. You can track this issue here: https://issuetracker.google.com/121348355.

Reply:
another commit resolved this problem.